components/feed/SaveFeedItemButton.tsx

```diff
@@ -0,0 +1,68 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import { useState, useTransition } from "react";
+
+import { saveFeedItem, unsaveFeedItem } from "@/lib/savedFeedItems";
+
+const SaveFeedItemButton = ({
+  userId,
+  postId,
+  savedItemIds,
+}: {
+  userId: string;
+  postId: string;
+  savedItemIds?: string[];
+}) => {
+  const saved = savedItemIds?.includes(postId.trim());
+
+  const [localSavedState, setLocalSavedState] = useState(saved);
+
+  const router = useRouter();
+  const [isPending, startTransition] = useTransition();
+  const [isFetching, setIsFetching] = useState(false);
+
+  // Create inline loading UI
+  const isMutating = isFetching || isPending;
+
+  const handleClick = async (e: React.MouseEvent) => {
+    e.preventDefault();
+    setIsFetching(true);
+    if (localSavedState) {
+      setLocalSavedState(!localSavedState);
+      await unsaveFeedItem(userId, postId);
+    } else {
+      setLocalSavedState(!localSavedState);
+      await saveFeedItem(userId, postId);
+    }
+    setIsFetching(false);
+    startTransition(() => {
+      // Refresh the current route and fetch new data from the server without
+      // losing client-side browser or React state.
+      router.refresh();
+    });
+  };
+
+  return (
+    <button type="button" onClick={(e) => void handleClick(e)}>
+      <svg
+        width="24"
+        height="24"
+        viewBox="0 0 24 24"
+        fill={localSavedState ? "white" : "black"}
```

Comment:
<img width="49" alt="Screenshot 2023-03-22 at 19 44 02" src="https://user-images.githubusercontent.com/7117993/227063233-dec72624-85f8-43d1-b177-9ec5c0a09a8d.png">
use DaisyUI classNames to style, instead of absolutes like black/white — in case someone changes the theme

https://daisyui.com/docs/colors/

Reply:
also can you make this button do the same loading thing as the filter (you can use className `btn btn-ghost` so it doesnt look like a button)